File: src/main/java/com/emarket/service/interfaz/CategoriaService.java
package com.emarket.service.interfaz;

import com.emarket.dto.categoria.CategoriaRequestDto;
import com.emarket.dto.categoria.CategoriaResponseDto;

import java.util.List;

public interface CategoriaService {
    CategoriaResponseDto crear(CategoriaRequestDto dto, String token);
    List<CategoriaResponseDto> listar();
    Double obtenerPrecioTotal(Long id);
    CategoriaResponseDto eliminar(Long id, String token);
}

File: frontend/src/services/categoria.service.ts
import { apiClient } from '../core/api/axiosInstance';
import type {
  CategoriaRequestDto,
  CategoriaResponseDto,
  ApiResponseDto,
} from '../core/types';

export const categoriaService = {
  async crear(dto: CategoriaRequestDto): Promise<CategoriaResponseDto> {
    const { data } = await apiClient.post<CategoriaResponseDto>('/categorias', dto);
    return data;
  },

  async listar(): Promise<CategoriaResponseDto[]> {
    const { data } = await apiClient.get<CategoriaResponseDto[]>('/categorias');
    return data;
  },

  async eliminar(id: number): Promise<ApiResponseDto<CategoriaResponseDto>> {
    const { data } = await apiClient.delete<ApiResponseDto<CategoriaResponseDto>>(
      `/categorias/eliminar/${id}`,
    );
    return data;
  },
};
